File: src/test/java/testdata/JsonIgnorePropertiesDocTestPOJO.java
package testdata;


import java.util.List;

public class JsonIgnorePropertiesDocTestPOJO {

    private String username;
    /**
     * @JsonIgnoreProperties users, aaa, bbb
     */
    private List<Role> roles;

    public class Role {

        private String roleName;
        private List<JsonIgnorePropertiesDocTestPOJO> users;
    }

}

File: src/test/java/testdata/kotlin/JsonIgnorePropertiesDocTestPOJO.kt
package testdata.kotlin

class JsonIgnorePropertiesDocTestPOJO {
    private val username: String? = null

    /**
     * @JsonIgnoreProperties users, aaa, bbb
     */
    private val roles: List<Role>? = null

    class Role {
        private val roleName: String? = null
        private val users: List<JsonIgnorePropertiesDocTestPOJO>? = null
    }
}